examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,391 @@
+        return False.
+    """
+    return Confirm.ask(
+        "Do you have a Google API key and configured? For example, GOOGLE_API_KEY=your-key and GOOGLE_CSE_ID=your-cse-id",
+        default=False,
+    )
+
+
+def run_debate() -> None:
+    """Execute the main debate logic.
+
+    Orchestrates the debate process, including setup, user input, LLM agent
+    interactions, and final feedback. Handles both user-guided and LLM-
+    delegated debates.
+
+    This function:
+    1. Loads global settings and the base LLM configurations.
+    2. Prompts the user to confirm if they want to use same LLM for all agents.
+    3. Prompts the user to select a debate topic and a side(Pro or Con).
+    4. Sets up pro, con, and feedback agents.
+    5. Runs the debate for a specified number of turns, either interactively
+       or autonomously.
+    6. Provides a feedback summary at the end.
+    7. Optionally validates references if a Google API key is configured. Creates a references validation agent.
+    Utilizes the langroid GoogleSearchTool.
+    """
+
+    global_settings = get_global_settings(nocache=True)
+    lr.utils.configuration.set_global(global_settings)
+    same_llm: bool = is_same_llm_for_all_agents()
+    llm_delegate: bool = is_llm_delegate()
+
+    # Get base LLM configuration
+    if same_llm:
+        shared_agent_config: OpenAIGPTConfig = get_base_llm_config("for main LLM")
+        pro_agent_config = con_agent_config = feedback_agent_config = shared_agent_config
+    else:
+        pro_agent_config: OpenAIGPTConfig = get_base_llm_config("for Pro Agent")
+        con_agent_config: OpenAIGPTConfig = get_base_llm_config("for Con Agent")
+        feedback_agent_config: OpenAIGPTConfig = get_base_llm_config("feedback and googleSearch")
+
+    system_messages: SystemMessages = load_system_messages("system_messages.json")
+    topic_name, pro_key, con_key, side = select_topic_and_setup_side(system_messages)
+
+    # Prompt for number of debate turns
+    max_turns: int = int(
+        Prompt.ask(f"How many turns should the debate continue for (Defalut is {DEFAULT_TURN_COUNT}?",
+                   default=DEFAULT_TURN_COUNT)
+    )
+
+    def create_chat_agent(
+            name: str,
+            llm_config: OpenAIGPTConfig,
+            system_message: str
+    ) -> ChatAgent:
+        """creates a ChatAgent with the given parameters.
+
+        Args:
+            name (str): The name of the agent.
+            llm_config (OpenAIGPTConfig): The LLM configuration for the agent.
+            system_message (str): The system message to guide the agent's LLM.
+
+        Returns:
+            ChatAgent: A configured ChatAgent instance.
+        """
+        return ChatAgent(
+            ChatAgentConfig(
+                llm=llm_config,
+                name=name,
+                system_message=system_message,
+            )
+        )
+
+    pro_agent = create_chat_agent("Pro", pro_agent_config, system_messages.messages[pro_key].message +
```

Comment:
here and in the next 3 lines, break the args into multiple lines for easier readability.